src/psyclone/tests/utilities_test.py

```diff
@@ -49,99 +56,237 @@
+        monkeypatch.setattr(Compile, "F90", "true")
+
     _compile = Compile(tmpdir)
-    test_compile = "psyclone.tests.utilities.Compile"
-    monkeypatch.setattr(test_compile+".TEST_COMPILE", False)
-    monkeypatch.setattr(test_compile+".TEST_COMPILE_OPENCL", False)
+
+    # Check that we always return True if compilation testing is disabled
+    monkeypatch.setattr(Compile, "TEST_COMPILE", False)
+    monkeypatch.setattr(Compile, "TEST_COMPILE_OPENCL", False)
     assert _compile.string_compiles("not fortran")
+
     # Re-enable compilation testing and check that we can build hello world
-    monkeypatch.setattr(test_compile+".TEST_COMPILE", True)
+    monkeypatch.setattr(Compile, "TEST_COMPILE", True)
     assert _compile.string_compiles(HELLO_CODE)
+
     # Repeat for some broken code
+    if Compile.F90 == "true":
+        monkeypatch.setattr(_compile, "_f90", "false")
     invalid_code = HELLO_CODE.replace("write", "wite", 1)
     assert not _compile.string_compiles(invalid_code)
 
 
+# -----------------------------------------------------------------------------
+def test_code_compile(change_into_tmpdir, monkeypatch):
+    # pylint: disable=unused-argument
+    '''A dummy test of the underlying code_compiles function, which takes
+    an AST. Note that the derived classes (GOceanBuild and LFRicBuild)
+    will test this properly (especially when compilation is enabled),
+    here we only to cover the other lines handling AST conversion. So,
+    here we just use 'true' as 'compiler'.
+    '''
+    _, invoke_info = parse(os.path.join(os.path.
+                                        dirname(os.path.
+                                                abspath(__file__)),
+                                        "test_files", "gocean1p0",
+                                        "single_invoke.f90"),
+                           api="gocean1.0")
+    psy = PSyFactory("gocean1.0", distributed_memory=False).create(invoke_info)
+    # Always trigger 'compilation' to avoid early abort, before the code
+    # to be tested is executed.
+    monkeypatch.setattr(Compile, "TEST_COMPILE", True)
+    monkeypatch.setattr(Compile, "F90", "true")
+    _compile = Compile()
+
+    # GOceanBuild or LFRicBuild will set the 'base_path' for the
+    # infrastructure files. We just need to set it to something:
+    _compile.base_path = os.getcwd()
+    # Only used to specify an existing dependency:
+    with open("hello_world.f90", "w", encoding="utf-8") as ffile:
+        ffile.write(HELLO_CODE)
+    assert _compile.code_compiles(psy, dependencies=["something.cl",
+                                                     "hello_world.f90"])
+    # Now check compilation failure, which we simulate by
+    # using 'false' as compiler:
+    monkeypatch.setattr(_compile, "_f90", "false")
+    assert _compile.code_compiles(psy, dependencies=["something.cl",
+                                                     "hello_world.f90"]) \
```

Comment:
Please use parentheses rather than `\`

Reply:
Done (I also fixed up some other lines in that file).